src/public/js/rs.js

```diff
@@ -21,6 +105,94 @@ var RS = (function() {
     }, '');
   }
 
+  // Encodes using systematic BCH Reed-Solomon on Galois Field 256
+  // Message should be passed in as length k block
+  // k specifies message length
+  // n specifies block length (number of symbols transmitted)
+  // n - k is the number of code symbols added.
+  // Encoding procedure has the largest order coefficient on the left
+  // and the smallest on the right. IE: [1,2,3] -> x^2 + 2x + 3
+  function encodeRSBlock(msg, n, k) {
+    remainder = polynomialDiv(genPoly, msg, n, k)
+    if (remainder == null) {
+      return null;
+    }
+    codedMsg = new Uint8Array(n)
+    for (i = 0; i < k; i++) {
+      codedMsg[i] = msg[i];
+    }
+    for (i = k; i < n; i++) {
+      codedMsg[i] = remainder[i - k]
+    }
+  }
+  function decodeRSBlock(codedMsg, n, k) {
+    remainder = polynomialDiv(genPoly, codedMsg, n, k)
+    if (remainder == null) {
+      return null;
```

Comment:
Yeah I was kind of using null as a pseudo-error type thing. Not sure the general pattern of errors in JS.